sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -294,6 +292,7 @@ private void initDataReader(Encoding dataEncoding, ByteBufferInputStream in) thr
 
   private void readPageV1(DataPageV1 page) throws IOException {
     this.pageValueCount = page.getValueCount();
+    this.pageFirstRowIndex = page.getFirstRowIndex().orElse(0L);
```

Comment:
same question as above, cannot `getFirstRowIndex` return 0? Or the index begins with 1?

Reply:
yea it can return 0, in which case it is the same as without column index and the whole page has to be read.